docs/source/compile/programming_model.md

```diff
@@ -0,0 +1,14 @@
+(programming_model)=
+# torch.compile Programming Model
+
+The `torch.compile` programming model
+1. clarifies some internal behaviors of `torch.compile` so that one can better predict compiler behavior on user code and
+2. provides ways for one to take more fine-grained control over `torch.compile`.
+
+By understanding the `torch.compile` programming model, one can systematically unblock themselves when encountering issues with `torch.compile`.
+
+```{toctree}
+:glob:
```

Comment:
```suggestion
:glob:
:maxdepth: 2
```
You likely want those to appear in the right nav under Section Navigation? Right now it's empty: https://docs-preview.pytorch.org/pytorch/pytorch/157985/compile/programming_model.html